test/prepare-manual-query.test.js

```diff
@@ -0,0 +1,16 @@
+var prepareManualQuery = require('../lib/prepare-manual-query'), assert = require('assert')
+
+describe('prepareManualQuery()', function () {
+
+  it('should limit the number of id\'s queried by the limit specified', function () {
+    var list = prepareManualQuery(
```

Comment:
I don't think this test is right, I think not using TypeScript has caught you out here.

The two arguments for `prepareManualQuery` are the list (which is an array) and the limit (which is a number). Here you've passed two arrays instead instead of an array and a number.

What you've got here has basically implemented the slicing outside of the method itself, so the test is testing the test and not the function.

It might be a bit easier to explain this on a call, but in a nutshell the test should look like this

```js
it('should limit the number of id\'s queried by the limit specified', function () {
    var q = prepareManualQuery(
      [ { _id: 0, shortTitle: 'James' }
      , { _id: 1, shortTitle: 'Robert' }
      ],
      1
    )
    assert.deepEqual(q.query._id.$in, [0])
```

Then you're testing what the method does

Reply:
Okay, yeah i see exactly what you mean. I have updated now.